src/OrchardCore.Modules/OrchardCore.ContentLocalization/Sitemaps/LocalizedContentItemsQueryProvider.cs

```diff
@@ -13,109 +13,120 @@ namespace OrchardCore.ContentLocalization.Sitemaps;
 
 public class LocalizedContentItemsQueryProvider : IContentItemsQueryProvider
 {
-    private readonly ISession _session;
+    private readonly IStore _store;
     private readonly IRouteableContentTypeCoordinator _routeableContentTypeCoordinator;
     private readonly ILocalizationService _localizationService;
 
     public LocalizedContentItemsQueryProvider(
-        ISession session,
+        IStore store,
         IRouteableContentTypeCoordinator routeableContentTypeCoordinator,
         ILocalizationService localizationService
         )
     {
-        _session = session;
+        _store = store;
         _routeableContentTypeCoordinator = routeableContentTypeCoordinator;
         _localizationService = localizationService;
     }
 
-    public async Task GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext queryContext)
+    public async Task<ContentItemsQueryResult> GetContentItemsAsync(ContentTypesSitemapSource source, ContentItemsQueryContext context)
     {
         var routeableContentTypeDefinitions = await _routeableContentTypeCoordinator.ListRoutableTypeDefinitionsAsync();
+        var session = _store.CreateSession(withTracking: false);
```

Comment:
```suggestion
        using var session = _store.CreateSession(withTracking: false);
```

The session can be disposed after the method is called. And even without tracking it will then release the items.

Reply:
100%